### PR TITLE
mbrola: init at 3.3

### PR DESCRIPTION
**Pull request by davidak**
###### Motivation for this change

espeak voices sound terrible. it might get a little better with mbrola

Closes https://github.com/NixOS/nixpkgs/pull/45580

###### Unsolved problems

espeak-ng don't find mbrola unless it's also installed. adding it as a dependency has not solved it

```
[davidak@ethmoid:~/code/nixpkgs]$ espeak -v mb/mb-de1 "ich mag züge"
mbrola: mbrola: No such file or directory
mbrowrap error: mbrola exited with status 1
Error: Could not load the specified mbrola voice file.
```

when i install mbrola, it don't finds the voices in `/home/davidak/.nix-profile/share/mbrola`. can we do anything about that except from waiting on upstream to solve it? https://github.com/numediart/MBROLA/issues/23

```
[davidak@ethmoid:~/code/nixpkgs]$ espeak -v mb/mb-de1 "ich mag züge"
mbrola: FATAL ERROR : cannot find file /usr/share/mbrola/voices/de1 !
mbrowrap error: mbrola closed stderr and did not exit
Error: Could not load the specified mbrola voice file.
```

maybe @jtojnar has some idea?

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by symphorien** [comment]
see also https://github.com/NixOS/nixpkgs/pull/45580

**Comment by jtojnar** [comment]
Yeah, the voices paths need to be solved in espeak, see the old PR.

File: pkgs/applications/audio/mbrola/default.nix
Comment (jtojnar) [inline pkgs/applications/audio/mbrola/default.nix:23]: When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`.

**Comment by davidak** [comment]
It works now!

Here is a demonstration of the german voices. (it's still very bad)

https://peertube.social/videos/watch/285ff782-3b68-45b4-92e6-e2c17cb642e8

I'm not sure if it's OK to enable mbrola by default in espeak-ng, since it adds 648 MB to the closure. But without it it is not really usable. @aske any opinion?

I don't see a way to package the voices separately, since the path is hardcoded in the patch.

File: pkgs/applications/audio/espeak-ng/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/espeak-ng/default.nix:3]:
```suggestion
, ronn, substituteAll
```

File: pkgs/applications/audio/mbrola/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/mbrola/default.nix:17]: How you do plan to update this? Maybe use a update script?

Comment (davidak) [reply pkgs/applications/audio/mbrola/default.nix:17]: i think about that when it is merged. probably manually

**Comment by davidak** [comment]
I resolved the merge conflict.

**Comment by SuperSandro2000** [comment]
Please do not merge master to resolve a merge conflict. This is the quick and dirty way but the proper way of rebasing and changing the line where the merge conflict is happening is required.

**Comment by davidak** [comment]
Sorry, didn't know GitHub does it that way. Will do it properly soon.

**Comment by davidak** [comment]
Now it's properly rebased. I have also tested it again.

```
builder for '/nix/store/l5wz7zhwvbnzkqyc24cgx1sciickh8yh-chromium-unwrapped-89.0.4343.0.drv' failed with exit code 1; last 10 log lines:
  tar: chromium-89.0.4343.0/third_party/libphonenumber/dist/resources/geocoding/pt/244.txt: Cannot write: No space left on device
  tar: chromium-89.0.4343.0/third_party/libphonenumber/dist/resources/geocoding/pt/245.txt: Cannot write: No space left on device
  tar: chromium-89.0.4343.0/third_party/libphonenumber/dist/resources/geocoding/pt/351.txt: Cannot write: No space left on device
  tar: chromium-89.0.4343.0/third_party/libphonenumber/dist/resources/geocoding/pt/239.txt: Cannot write: No space left on device
  tar: chromium-89.0.4343.0/third_party/libphonenumber/dist/resources/geocoding/pt/55.txt: Cannot write: No space left on device
  tar: chromium-89.0.4343.0/third_party/libphonenumber/dist/resources/geocoding/de/32.txt: Cannot write: No space left on device
  tar: chromium-89.0.4343.0/third_party/libphonenumber/dist/resources/geocoding/de/49.txt: Cannot write: No space left on device
  tar: chromium-89.0.4343.0/third_party/libphonenumber/dist/resources/geocoding/de/41.txt: Cannot write: No space left on device
  tar: Exiting with failure status due to previous errors
  do not know how to unpack source archive /nix/store/3ywbdy67xqisia5k3i7y8f0l0dr2p3c9-chromium-89.0.4343.0.tar.xz
cannot build derivation '/nix/store/i3hqazyipf7hng1mkbdhbk4fnv5agw14-chromium-dev-89.0.4343.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/3728zpgs9viid2dphlyviczpk1j7nvdh-env.drv': 6 dependencies couldn't be built
[32 built (6 failed), 560 copied (7160.8 MiB), 3410.5 MiB DL]
error: build of '/nix/store/3728zpgs9viid2dphlyviczpk1j7nvdh-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/93401
6 packages failed to build:
chromium chromiumBeta chromiumDev navit tts ungoogled-chromium

13 packages built:
dasher direwolf espeak gnome3.orca google-chrome google-chrome-beta google-chrome-dev mbrola pantheon.elementary-session-settings python37Packages.phonemizer python38Packages.phonemizer python39Packages.phonemizer speechd
```

**Comment by davidak** [comment]
I rebased and tested again.

@SuperSandro2000 can you merge quickly before it diverges again?

To test:

```
[davidak@gaming:~/code/nixpkgs]$ nix run -f /home/davidak/code/nixpkgs/ espeak

[davidak@gaming:~/code/nixpkgs]$ for i in {1..8}; do echo "mb/mb-de$i"; espeak -v mb/mb-de$i "Er hat übermächtige Käsefüße!"; done
mb/mb-de1
mb/mb-de2
mb/mb-de3
mb/mb-de4
mb/mb-de5
mb/mb-de6
mb/mb-de7
mb/mb-de8
```

You should hear different voices.

File: pkgs/applications/audio/mbrola/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/mbrola/default.nix:14]: Generally we move second fetches into a let in.

File: pkgs/applications/audio/espeak-ng/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/espeak-ng/default.nix:3]: The voices are pretty big. Maybe we should disable it by default.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/espeak-ng/default.nix
Comment (jtojnar) [reply pkgs/applications/audio/espeak-ng/default.nix:3]: The improvement is very noticeable and it’s not like TTS is enabled anywhere by default. I would just leave it in for now and focus on writing a patch for espeak for finding mbrola voices in `XDG_DATA_DIRS` when I have some time.

Comment (davidak) [reply pkgs/applications/audio/espeak-ng/default.nix:3]: @jtojnar yes, that would be perfect. then one can install just the language they need

File: pkgs/applications/audio/mbrola/default.nix
Comment (davidak) [reply pkgs/applications/audio/mbrola/default.nix]: like this?

File: pkgs/applications/audio/mbrola/default.nix
Comment (jtojnar) [inline pkgs/applications/audio/mbrola/default.nix:30]: Instead of inheriting above, you can just use `${voices}` here. Also instead of copying, you can just symlink to avoid slow copying.

File: pkgs/applications/audio/mbrola/default.nix
Comment (jtojnar) [inline pkgs/applications/audio/mbrola/default.nix:1]: CI seems to fail because you are requesting pkgconfig instead of pkg-config. But most of these are not used anyway.

Comment (davidak) [reply pkgs/applications/audio/mbrola/default.nix]: thanks, fixed

i think i added all those because they are needed for the build to succeed. i don't remember...

File: pkgs/applications/audio/mbrola/default.nix
Comment (davidak) [reply pkgs/applications/audio/mbrola/default.nix]: done

File: pkgs/applications/audio/mbrola/default.nix
Comment (jtojnar) [reply pkgs/applications/audio/mbrola/default.nix]: Well, you are not using them in inputs any more so they can be removed.

Comment (davidak) [reply pkgs/applications/audio/mbrola/default.nix]: you are right, i tried it

**Review by jtojnar — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 93401` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>navit: log was empty</li>
  </ul>
</details>
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>chromium: log was empty</li>
<li>puppeteer-cli: log was empty</li>
<li>ungoogled-chromium: log was empty</li>
  </ul>
</details>
<details>
<summary>
13 packages built:
</summary>
<ul>
<li>dasher</li>
<li>direwolf</li>
<li>espeak (espeak-ng)</li>
<li>orca (gnome3.orca)</li>
<li>google-chrome</li>
<li>google-chrome-beta</li>
<li>google-chrome-dev</li>
<li>mbrola</li>
<li>pantheon.elementary-session-settings</li>
<li>python37Packages.phonemizer</li>
<li>python38Packages.phonemizer</li>
<li>python39Packages.phonemizer</li>
<li>speechd</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
dasher:
<details>


`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/accessibility/dasher/default.nix:50:5:

```
   |
50 |     license = lib.licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
direwolf:
<details>


`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/radio/direwolf/default.nix:44:5:

```
   |
44 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
perl is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/radio/direwolf/default.nix:19:3:

```
   |
19 |   buildInputs = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/radio/direwolf/default.nix:24:5:

```
   |
24 |     ./udev-fix.patch
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
Unused argument: `glibc`.
Near pkgs/applications/radio/direwolf/default.nix:1:57:

```
  |
1 | { lib, stdenv, fetchFromGitHub, cmake, alsaLib, espeak, glibc, gpsd
  |                                                         ^
```
</details>
google-chrome-beta:
<details>


`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/networking/browsers/google-chrome/default.nix:94:3:

```
   |
94 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
google-chrome-dev:
<details>


`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/networking/browsers/google-chrome/default.nix:94:3:

```
   |
94 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
google-chrome:
<details>


`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/networking/browsers/google-chrome/default.nix:94:3:

```
   |
94 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
mbrola:
<details>


Unused argument: `autoconf`.
Near pkgs/applications/audio/mbrola/default.nix:1:33:

```
  |
1 | { stdenv, lib, fetchFromGitHub, autoconf, automake, which, libtool, pkg-config }:
  |                                 ^
```
Unused argument: `automake`.
Near pkgs/applications/audio/mbrola/default.nix:1:43:

```
  |
1 | { stdenv, lib, fetchFromGitHub, autoconf, automake, which, libtool, pkg-config }:
  |                                           ^
```
Unused argument: `which`.
Near pkgs/applications/audio/mbrola/default.nix:1:53:

```
  |
1 | { stdenv, lib, fetchFromGitHub, autoconf, automake, which, libtool, pkg-config }:
  |                                                     ^
```
Unused argument: `libtool`.
Near pkgs/applications/audio/mbrola/default.nix:1:60:

```
  |
1 | { stdenv, lib, fetchFromGitHub, autoconf, automake, which, libtool, pkg-config }:
  |                                                            ^
```
Unused argument: `pkg-config`.
Near pkgs/applications/audio/mbrola/default.nix:1:69:

```
  |
1 | { stdenv, lib, fetchFromGitHub, autoconf, automake, which, libtool, pkg-config }:
  |                                                                     ^
```
</details>
navit:
<details>


`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/misc/navit/default.nix:86:5:

```
   |
86 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/misc/navit/default.nix:87:5:

```
   |
87 |     maintainers = [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
error: no-flags-spaces
`cmakeFlags` cannot contain spaces, please use `cmakeFlagsArray` in Bash.

Near pkgs/applications/misc/navit/default.nix:43:3:

```
   |
43 |   cmakeFlags = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-spaces.md
Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/misc/navit/default.nix:36:15:

```
   |
36 |   patches = [ ./CMakeLists.txt.patch ];
   |               ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
orca:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/misc/orca/default.nix:48:5:

```
   |
48 |     (substituteAll {
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
pantheon.elementary-session-settings:
<details>


`lgpl3` is a deprecated license, check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/desktops/pantheon/desktop/elementary-session-settings/default.nix:159:5:

```
    |
159 |     license = licenses.lgpl3;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
Unused argument: `substituteAll`.
Near pkgs/desktops/pantheon/desktop/elementary-session-settings/default.nix:4:3:

```
  |
4 | , substituteAll
  |   ^
```
Unused argument: `at-spi2-core`.
Near pkgs/desktops/pantheon/desktop/elementary-session-settings/default.nix:14:3:

```
   |
14 | , at-spi2-core
   |   ^
```
</details>
puppeteer-cli:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:359:7:

```
    |
359 |       meta = {
    |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
rsync is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:292:7:

```
    |
292 |       buildInputs = [ yarn nodejs rsync ] ++ extraBuildInputs;
    |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
yarn is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:292:7:

```
    |
292 |       buildInputs = [ yarn nodejs rsync ] ++ extraBuildInputs;
    |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
Unused argument: `version`.
Near pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:67:5:

```
   |
67 |     version,
   |     ^
```
</details>
python37Packages.phonemizer:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/python-modules/phonemizer/default.nix:36:5:

```
   |
36 |     ./remove-intertwined-festival-test.patch
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
Unused argument: `python3Packages`.
Near pkgs/development/python-modules/phonemizer/default.nix:5:3:

```
  |
5 | , python3Packages
  |   ^
```
Unused argument: `pkgs`.
Near pkgs/development/python-modules/phonemizer/default.nix:6:3:

```
  |
6 | , pkgs
  |   ^
```
Unused argument: `pytestrunner`.
Near pkgs/development/python-modules/phonemizer/default.nix:12:3:

```
   |
12 | , pytestrunner
   |   ^
```
</details>
python38Packages.phonemizer:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/python-modules/phonemizer/default.nix:36:5:

```
   |
36 |     ./remove-intertwined-festival-test.patch
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
Unused argument: `python3Packages`.
Near pkgs/development/python-modules/phonemizer/default.nix:5:3:

```
  |
5 | , python3Packages
  |   ^
```
Unused argument: `pkgs`.
Near pkgs/development/python-modules/phonemizer/default.nix:6:3:

```
  |
6 | , pkgs
  |   ^
```
Unused argument: `pytestrunner`.
Near pkgs/development/python-modules/phonemizer/default.nix:12:3:

```
   |
12 | , pytestrunner
   |   ^
```
</details>
python39Packages.phonemizer:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/python-modules/phonemizer/default.nix:36:5:

```
   |
36 |     ./remove-intertwined-festival-test.patch
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
Unused argument: `python3Packages`.
Near pkgs/development/python-modules/phonemizer/default.nix:5:3:

```
  |
5 | , python3Packages
  |   ^
```
Unused argument: `pkgs`.
Near pkgs/development/python-modules/phonemizer/default.nix:6:3:

```
  |
6 | , pkgs
  |   ^
```
Unused argument: `pytestrunner`.
Near pkgs/development/python-modules/phonemizer/default.nix:12:3:

```
   |
12 | , pytestrunner
   |   ^
```
</details>
speechd:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/libraries/speechd/default.nix:50:5:

```
   |
50 |     (substituteAll {
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
</details>

File: pkgs/applications/audio/espeak-ng/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/espeak-ng/default.nix:1]:
```suggestion
{ stdenv, lib, fetchFromGitHub
```

all unused

Comment (davidak) [reply pkgs/applications/audio/espeak-ng/default.nix]: don't forget to squash when merging. github can do that

Comment (SuperSandro2000) [reply pkgs/applications/audio/espeak-ng/default.nix]: Sorry that was wrong. @jtojnar you have an idea what hammering did here?

mbrola:
<details>


Unused argument: `autoconf`.
Near pkgs/applications/audio/mbrola/default.nix:1:33:

```
  |
1 | { stdenv, lib, fetchFromGitHub, autoconf, automake, which, libtool, pkg-config }:
  |                                 ^
```
Unused argument: `automake`.
Near pkgs/applications/audio/mbrola/default.nix:1:43:

```
  |
1 | { stdenv, lib, fetchFromGitHub, autoconf, automake, which, libtool, pkg-config }:
  |                                           ^
```
Unused argument: `which`.
Near pkgs/applications/audio/mbrola/default.nix:1:53:

```
  |
1 | { stdenv, lib, fetchFromGitHub, autoconf, automake, which, libtool, pkg-config }:
  |                                                     ^
```
Unused argument: `libtool`.
Near pkgs/applications/audio/mbrola/default.nix:1:60:

```
  |
1 | { stdenv, lib, fetchFromGitHub, autoconf, automake, which, libtool, pkg-config }:
  |                                                            ^
```
Unused argument: `pkg-config`.
Near pkgs/applications/audio/mbrola/default.nix:1:69:

```
  |
1 | { stdenv, lib, fetchFromGitHub, autoconf, automake, which, libtool, pkg-config }:
  |                                                                     ^
```
</details>

Comment (jtojnar) [reply pkgs/applications/audio/espeak-ng/default.nix]: That warnings originated in mbrola, which already has been fixed.

**Comment by jtojnar** [comment]
Thanks.

File: pkgs/applications/audio/espeak-ng/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/audio/espeak-ng/default.nix:1]: I need to add upstream the commit link to nixpkgs-review.

**Pull request merged**
